src/platform/mbed/ConnectivityManagerImpl.cpp

```diff
@@ -0,0 +1,496 @@
+        frequency = 5785;
+        break;
+    case 161:
+        frequency = 5805;
+        break;
+    case 165:
+        frequency = 5825;
+        break;
+    }
+
+    return frequency;
+}
+
+static uint16_t MapFrequency(const uint16_t inBand, const uint8_t inChannel)
+{
+    uint16_t frequency = 0;
+
+    if (inBand == WIFI_BAND_2_4GHZ)
+    {
+        frequency = Map2400MHz(inChannel);
+    }
+    else if (inBand == WIFI_BAND_5_0GHZ)
+    {
+        frequency = Map5000MHz(inChannel);
+    }
+
+    return frequency;
+}
+
+void ConnectivityManagerImpl::GetWifiStatus(void)
+{
+    _interface  = WiFiInterface::get_default_instance();
+    auto status = _interface->get_connection_status();
+    ChipLogDetail(DeviceLayer, "Connection status: %s", status2str(status));
+    ChipLogDetail(DeviceLayer, "MAC: %s", _interface->get_mac_address());
+    SocketAddress a;
+    _interface->get_ip_address(&a);
+    ChipLogDetail(DeviceLayer, "IP: %s", a.get_ip_address());
+    _interface->get_netmask(&a);
+    ChipLogDetail(DeviceLayer, "Netmask: %s", a.get_ip_address());
+    _interface->get_gateway(&a);
+    ChipLogDetail(DeviceLayer, "Gateway: %s", a.get_ip_address());
+    ChipLogDetail(DeviceLayer, "RSSI: %d", _interface->get_rssi());
+}
+
+// ==================== ConnectivityManager Platform Internal Methods ====================
+
+CHIP_ERROR ConnectivityManagerImpl::_Init()
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    mWiFiStationMode                = kWiFiStationMode_Disabled;
+    mWiFiStationState               = kWiFiStationState_NotConnected;
+    mIsProvisioned                  = false;
+    mWiFiAPMode                     = kWiFiAPMode_NotSupported;
+    mWiFiStationReconnectIntervalMS = CHIP_DEVICE_CONFIG_WIFI_STATION_RECONNECT_INTERVAL;
+    mWiFiAPIdleTimeoutMS            = CHIP_DEVICE_CONFIG_WIFI_AP_IDLE_TIMEOUT;
+
+    // TODO Initialize the Chip Addressing and Routing Module.
+
+    return err;
+}
+
+void ConnectivityManagerImpl::_OnPlatformEvent(const ChipDeviceEvent * event) {}
+
+void ConnectivityManagerImpl::OnInterfaceEvent(nsapi_event_t event, intptr_t data)
+{
+    if (event == NSAPI_EVENT_CONNECTION_STATUS_CHANGE)
+    {
+        switch (data)
+        {
+        case NSAPI_STATUS_LOCAL_UP:
+            break;
+        case NSAPI_STATUS_GLOBAL_UP:
+            OnStationConnected();
+            break;
+        case NSAPI_STATUS_DISCONNECTED:
+            OnStationDisconnected();
+            break;
+        case NSAPI_STATUS_CONNECTING:
+            break;
+        default:
+
+            break;
+        }
+    }
+}
+
+CHIP_ERROR ConnectivityManagerImpl::ProvisionWiFiNetwork(const char * ssid, const char * key)
+{
+    _interface = WiFiInterface::get_default_instance();
+
+    ChipLogDetail(DeviceLayer, "connect to %s with %s password", ssid, key);
+    _interface->attach(mbed::callback(this, &ConnectivityManagerImpl::OnInterfaceEvent));
+
+    auto result = _interface->connect(ssid, key, NSAPI_SECURITY_WPA_WPA2);
```

Comment:
I think this could be two steps: 
- Scan to establish the security type 
- Connect with the right security setting.